### PR TITLE
gigasecond: add immutable test

### PR DESCRIPTION
**Pull request by SleeplessByte**

### CODE REVIEW

File: canonical-schema.json [hunk 1]
```diff
@@ -1,158 +1,200 @@
...
-               { "error": { "maxProperties": 1 }
-               }
-          },
-
-      "error":
-          { "description": "A message describing an error condition"
-          , "type"       : "string"
-          }
-
+  "comments": [
```
Comment (SaschaMann) [inline canonical-schema.json:2]: This (and the last few PRs) all changed the schema formatting. Is that intended?

Comment (SleeplessByte) [reply canonical-schema.json:2]: That's what prettier is doing. Schema isn't ignored.

File: exercises/gigasecond/canonical-data.json [hunk 2]
```diff
@@ -62,6 +62,23 @@
...
+          "description": "does not mutate the input",
+          "comments": [
+            "This test case is for languages with a mutable date or datetime  ",
+            "datastructure. After running add(moment), moment should remain   ",
+            "the same."
+          ],
+          "scenarios": [
+            "immutable"
+          ],
+          "property": "moment",
```
Comment (ErikSchierboom) [inline exercises/gigasecond/canonical-data.json:77]: As the property is now no longer the method to be called, you could consider using a descriptive property like the one used in the [diffie-hellman exercise](https://github.com/exercism/problem-specifications/blob/master/exercises/diffie-hellman/canonical-data.json#L21). That said, the current property does work too, so feel free to ignore.

**Review by ErikSchierboom — APPROVED** [review]

File: exercises/gigasecond/canonical-data.json (hunk 2 above)
Comment (SleeplessByte) [inline exercises/gigasecond/canonical-data.json:77]:
```suggestion
          "property": "isEqual",
```

**Review by ErikSchierboom — APPROVED** [review]
Nice!

**Comment by ErikSchierboom** [comment]
@SleeplessByte Could you fix the merge conflicts?

**Review by kytrinyx — APPROVED** [review]
This is a nice change. @SleeplessByte do you have the time to fix the conflict? If not holler and I'll get it sorted.

**Comment by SleeplessByte** [comment]
Eh. I'll make time today :)

**Comment by SleeplessByte** [comment]
@SaschaMann wanna approve it now that the format is gone?

**Review by SaschaMann — APPROVED** [review]

**Pull request merged**